Core/include/Acts/Surfaces/ConeSurface.hpp

```diff
@@ -83,12 +83,6 @@ class ConeSurface : public Surface {
   /// @param other is the source surface for the assignment
   ConeSurface& operator=(const ConeSurface& other);
 
-  /// Clone method into a concrete type of ConeSurface with shift
-  /// @param gctx The current geometry context object, e.g. alignment
-  /// @param shift applied to the surface
-  std::shared_ptr<ConeSurface> clone(const GeometryContext& gctx,
```

Comment:
Ok, so the only reason I can think of why we'd want to have a `clone()` method is this: If you have a surface in a shared_ptr to the **base** class, if you want to make a copy, you have to dynamic_cast to the concrete **derived** class in order to be able to copy it. That can be quite verbose! This implementation here uses the virtual dispatch mechanism to do it.

So, are we **sure** this is not something that we need occasionally? Especially when we start more serious integration into experiments?

Reply:
Making a `copy/clone` of a `shared_ptr` surface does not make sense, we should just assign the `shared_ptr`.

The fact that we didn't use it for 3 odd years should answer the question. The reason why we had in in the ATLAS code was that we had some manual memory bookkeeping.

Reply:
Ok fair enough.